src/modules/blockchainApplication/explore/components/BlockchainApplicationRow/BlockchainApplicationRow.js

```diff
@@ -28,48 +31,73 @@ const ChainStatus = ({ status, t }) => (
 
 const ChainName = ({ title, logo }) => (
   <div className={`chain-name ${grid['col-xs-4']} ${styles.chainName}`}>
-    <img src={logo} />
+    {logo ? (
+      <img src={logo} />
+    ) : (
+      <div className={styles.chainInitials}>
+        {title?.[0]?.toUpperCase()}
+        <CautionIcon />
+      </div>
+    )}
     <span>{title}</span>
   </div>
 );
 
-const Pin = ({ isPinned, onTogglePin }) => (
-  <div className={`${styles.pinWrapper} ${isPinned ? styles.show : ''}`}>
-    <TertiaryButton onClick={onTogglePin} className="blockchain-application-pin-button">
-      <Icon data-testid="pin-button" name={isPinned ? 'pinnedIcon' : 'unpinnedIcon'} />
-    </TertiaryButton>
-  </div>
-);
-
 const BlockchainApplicationRow = ({ data, className, t }) => {
-  const { checkPinByChainId, togglePin } = usePinBlockchainApplication();
-
-  const handleTogglePin = useCallback(
-    (event) => {
-      event.stopPropagation();
-      togglePin(data.chainID);
-    },
-    [togglePin]
-  );
+  const { checkPinByChainId } = usePinBlockchainApplication();
 
   const application = {
     ...data,
     isPinned: checkPinByChainId(data.chainID),
   };
 
-  return (
-    <div data-testid="applications-row" className={`application-row ${styles.container}`}>
-      <DialogLink
+  const registerApplication = () => {
+    window.open(sideChainRegistrationLink, '_blank');
+  };
+
+  const showFlashMessage = () => {
+    FlashMessageHolder.addMessage(
+      <WarnMissingAppMetaData registerApplication={registerApplication} />,
+      'WarnMissingAppMetaData'
+    );
+  };
+
+  const RowWrapper = ({ children }) => {
```

Comment:
It is always better to not inline components. If you move this outside of the component and pass in application and className we can prevent this side effect (the flickering of the images in the rows).

https://github.com/LiskHQ/lisk-desktop/assets/8784876/869388d9-04d9-4200-b733-1bc992fca18d